File: PHP/Pc.php
<?php
include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc
{
    // private atribute dari kelas Pc
    private $processorObj;
    private $diskObj;
    private $ramObj;
    private $totalPrice;

    // konstruktor
    function __construct($processorObj, $diskObj, $ramObj)
    {
        $this->processorObj = $processorObj;
        $this->diskObj = $diskObj;
        $this->ramObj = $ramObj;
    }

    // mengeset nilai atribut processor
    function  setProcessorObj($processorObj)
    {
        $this->$processorObj = $processorObj;
    }

    // mengeset nilai atribut disk
    function  setDiskObj($diskObj)
    {
        $this->$diskObj = $diskObj;
    }

    // mengeset nilai atribut ram
    function  setRamObj($ramObj)
    {
        $this->ramObj = $ramObj;
    }

    // mengeset nilai atribut totalPrice
    function  setTotalPrice($processorPrice, $diskPrice, $ramPrice)
    {
        $this->totalPrice = $processorPrice + $diskPrice + $ramPrice;
    }

    // mengembalikan nilai atribut processor
    function getProcessorObj()
    {
        return $this->processorObj;
    }

    // mengembalikan nilai atribut disk
    function getDiskObj()
    {
        return $this->diskObj;
    }

    // mengembalikan nilai atribut ram
    function getRamObj()
    {
        return $this->ramObj;
    }

    // mengembalikan nilai atribut totalPrice
    function getTotalPrice()
    {
        return $this->totalPrice;
    }

    // menampilkan atribut Pc
    function  printPc()
    {
        $this->processorObj->printProcessor();
        $this->diskObj->printDisk();
        $this->ramObj->printRam();
        $this->setTotalPrice($this->processorObj->getPrice(), $this->diskObj->getPrice(), $this->ramObj->getPrice());
        echo "Total Price: " . $this->getTotalPrice() . "<br>";
    }

    function __destruct()
    {
    }
}

File: Python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram


class Pc:
    # private atribute dari kelas Pc
    __processorObj = Processor()
    __diskObj = Disk()
    __ramObj = Ram()
    __totalPrice = 0

    # konstruktor
    def __init__(self, processorObj, diskObj, ramObj):
        self.__processorObj = processorObj
        self.__diskObj = diskObj
        self.__ramObj = ramObj

    # mengeset nilai atribut processor
    def setProcessorObj(self, processorObj):
        self.__processorObj = processorObj

    # mengeset nilai atribut disk
    def setDiskObj(self, diskObj):
        self.__diskObj = diskObj

    # mengeset nilai atribut ram
    def setRamObj(self, ramObj):
        self.__ramObj = ramObj

    # mengeset nilai atribut totalPrice
    def setTotalPrice(self, processorPrice, diskPrice, ramPrice):
        self.__totalPrice = processorPrice + diskPrice + ramPrice

    # mengembalikan nilai atribut processor
    def getProcessorObj(self):
        return self.__processorObj

    # mengembalikan nilai atribut disk
    def getDiskObj(self):
        return self.__diskObj

    # mengembalikan nilai atribut ram
    def getRamObj(self):
        return self.__ramObj

    # mengembalikan nilai atribut totalPrice
    def getTotalPrice(self):
        return self.__totalPrice

    # menampilkan atribut Pc
    def printPc(self):
        self.__processorObj.printProcessor()
        self.__diskObj.printDisk()
        self.__ramObj.printRam()
        self.setTotalPrice(self.__processorObj.getPrice(),
                           self.__diskObj.getPrice(), self.__ramObj.getPrice())
        print("Total Price: " + str(self.getTotalPrice()))
